### DISCUSSION TITLE
Feedback requested on plan to create new tutorials

### DISCUSSION THREAD

**Discussion by utsab** [AcademySoftwareFoundation/OpenTimelineIO#930] [Ideas]
I am working with @hisergiorojas to create additional tutorials for OTIO.  After reading through the available documentation, we outlined a few use cases for OTIO that have not been covered in the [examples](https://github.com/PixarAnimationStudios/OpenTimelineIO/tree/master/examples) folder: 

- Nested compositions (flatten them)
- Illustrate a transition with and without a gap. 
- Illustrate a transition between different kinds of Composable objects (clip to a track, track to a stack) 
- Multiple tracks (summarize them, flatten them) 

We have two questions: 

1. Are the outlined scenarios good examples of how to use OTIO?  If you can suggest better examples, we'd be happy to hear your feedback.   
2. Our plan is start by creating code samples for the above scenarios and contribute them to the  [examples](https://github.com/PixarAnimationStudios/OpenTimelineIO/tree/master/examples) folder.  What do you think of this plan?  

Thank you.

**Comment by meshula** [comment]
I think organizing some examples would be very helpful. My suggestion is that we use the umbrella organization https://github.com/OpenTimelineIO/ as a place to hold an actual examples repo, and that we work to trim the library repo down to the minimum for building the library and the artifacts that are derived from the source code as a consequence of the build, such as the documentation. Within an examples repo, there could be individual directories, one per language binding for which examples are available.

**Reply by utsab** [reply]
Hi @meshula, thanks for your feedback.  I think it makes sense to create a separate repo specifically for examples.  What are the next steps for this?

**Reply by meshula** [reply]
Maybe come to the next TSC meeting, and introduce your proposal and get some feedback?

**Reply by utsab** [reply]
Sure, that sounds like a great idea.  @hisergiorojas and I will plan on attending the next meeting which I believe is on April 29.

**Comment by jminor** [comment]
These are great examples for a tutorial. Do you intend the tutorials to be in Python or C++ or both?

Some of these use cases are covered by the `flatten_stack` method in the API. (See example here: https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/examples/flatten_video_tracks.py ) and some are not. I am guessing that we will be tempted to add features to that flatten method to make the tutorial smoother - which is a great outcome for this also :)

**Reply by utsab** [reply]
@jminor Thanks for your feedback.  We were intending to write the tutorials in Python, but we're open to hearing feedback on this.  If there is a need for additional C++ examples, we'd be happy to try to contribute there as well.  What do you think?  Which language is most in need of more examples?

**Reply by jminor** [reply]
Python examples are great since most people trying out OTIO for the first time will be more familiar with it. Eventually we should have some C++ tutorials also, as I think @darbyjohnston might have suggested in another thread, but that could be done separately.

**Reply by darbyjohnston** [reply]
If you are interested there are a couple C++ examples here:
https://github.com/PixarAnimationStudios/OpenTimelineIO/tree/master/examples

I tried to match the corresponding Python code so they could be easily compared. I also have a C++ version of the "conform.py" example but it's waiting on some other code to be checked in first.
